var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -65,6 +65,7 @@ class Silo(AutotoolsPackage):
     variant("hzip", default=True, description="Enable hzip support")
     variant("fpzip", default=True, description="Enable fpzip support")
 
+    depends_on("python", type="build", when="+python")
```

Comment:
Does this not need python at run time? Why does it need python.h if not to then link against it?

See https://github.com/LLNL/Silo/blob/f57b12f1faaf9699ad86a60602cd9a65689585cc/tools/python/CMakeLists.txt#L69

```suggestion
    depends_on("python", type=("build", "link"), when="+python")
```
And move to where other "regular" dependencies are.